src/main/java/it/gov/pagopa/pu/sil/service/querypayments/PaaSILChiediPosizioniAperteService.java

```diff
@@ -0,0 +1,156 @@
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class PaaSILChiediPosizioniAperteService {
+
+  private static final String OBJECT_VERSION = "6.2.0";
+
+  private final DebtPositionService debtPositionService;
+  private final OrganizationService organizationService;
+  private final JAXBTransformService jaxbTransformService;
+  private final CheckoutClient checkoutClient;
+  private final CartRequestMapper cartRequestMapper;
+  private final DebtPositionTypeService debtPositionTypeService;
+
+  public PaaSILChiediPosizioniAperteRisposta processRequest(
+    PaaSILChiediPosizioniAperte request,
+    UserInfo userInfo,
+    String accessToken,
+    String orgIpaCode
+  ) {
+    PaaSILChiediPosizioniAperteRisposta response = new PaaSILChiediPosizioniAperteRisposta();
+    AuthorizationService.validateAdminRole(request.getCodIpaEnte(), userInfo);
+
+    Long organizationId = AuthorizationService.getOrganizationIdFromUserInfo(userInfo, request.getCodIpaEnte());
+    Organization organization = organizationService.getOrganizationById(organizationId, accessToken)
+      .orElse(null);
+    if (organization == null || !OrganizationStatus.ACTIVE.equals(organization.getStatus())) {
+      throw new SilFaultException(SilFaults.PAA_ENTE_NON_VALIDO, "L'ente non è valido o non è abilitato");
+    }
+
+    List<DebtPositionDTO> debtPositions = debtPositionService.getDebtPositionsByDebtorFiscalCodeAndDebtorEntityType(
+      request.getIdentificativoUnivocoPersonaFG().getCodiceIdentificativoUnivoco(),
+      PersonEntityType.fromValue(request.getIdentificativoUnivocoPersonaFG().getTipoIdentificativoUnivoco().value()),
+      organizationId,
+      InstallmentStatus.UNPAID,
+      null,
+      null,
+      accessToken
+    );
+
+    response.getPaaSILPosizioniApertes().addAll(processOpenPositions(request, organization, debtPositions, accessToken));
+    return response;
+  }
+
+  private List<PaaSILPosizioniAperte> processOpenPositions(PaaSILChiediPosizioniAperte request, Organization organization, List<DebtPositionDTO> debtPositions, String accessToken) {
+    return debtPositions.stream()
+      .flatMap(debtPosition -> debtPosition.getPaymentOptions().stream()
+        .flatMap(paymentOption -> paymentOption.getInstallments().stream()
+          .map(installment -> {
+            PaaSILPosizioniAperte openPosition = new PaaSILPosizioniAperte();
+            openPosition.setCodIpaEnte(request.getCodIpaEnte());
+            openPosition.setDeNomeEnte(organization.getOrgName());
+
+            try {
+              CartRequest cartRequest = cartRequestMapper.mapInstallmentToCartRequest(installment, organization, null, null);
+              openPosition.setUrlPagamento(checkoutClient.checkoutCart(cartRequest));
+            } catch (Exception e) {
+              log.error("Error generating payment URL for installment[{}]", installment.getInstallmentId(), e);
+            }
```

Comment:
as discussed before, this URL should not be checkout URL...
@robertopasetti here we should set a TODO to remember to set ARpu URL instead?

see: https://github.com/pagopa/p4pa-pu-sil/pull/158/files/de5de34038537692d64b86f18eae87a7f2ce6ec5#r2416044815